src/core/thread/mle.cpp

```diff
@@ -1251,6 +1251,18 @@ void Mle::HandleNotifierEvents(Events aEvents)
     }
 #endif
 
+    if (aEvents.Contains(kEventSupportedChannelMaskChanged))
+    {
+        Mac::ChannelMask channelMask = Get<Mac::Mac>().GetSupportedChannelMask();
+
+        if (!channelMask.ContainsChannel(Get<Mac::Mac>().GetPanChannel()) && (mRole != kRoleDisabled))
+        {
+            LogCrit("Channel %u is not in the supported channel mask %s, detach the network gracefully!",
```

Comment:
I think `Crit` indicates the process is about to die.
```suggestion
            LogWarn("Channel %u is not in the supported channel mask %s, detach the network gracefully!",
```

Reply:
Updated